aead/src/lib.rs

```diff
@@ -0,0 +1,214 @@
+//! A set of traits designed to support authenticated encryption.
+
+#![no_std]
+
+extern crate alloc;
```

Comment:
This precludes usage on platforms without a heap. It'd be nice to gate this on an `alloc` feature and make the core APIs `no_std`-friendly with allocating wrappers. See Misreant's `seal_in_place`:

https://github.com/miscreant/miscreant.rs/blob/master/src/aead.rs#L57

...vs its `seal` wrapper which allocates:

https://github.com/miscreant/miscreant.rs/blob/master/src/aead.rs#L74